game-app/game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AvailableSupports.java

```diff
@@ -149,8 +150,11 @@ int giveSupportToUnit(final Unit unit) {
     return amountOfSupportGiven;
   }
 
-  private int getSupportAvailable(final UnitSupportAttachment support) {
-    return Math.max(0, Math.min(support.getBonusType().getCount(), getSupportLeft(support)));
+  /** change to retrun the number of supporters (units) not supports */
+  private int getSupportersAvailable(final UnitSupportAttachment support) {
+    final SupportDetails details = supportUnits.get(support);
+    final IntegerMap<Unit> intMap = details.supportUnits;
+    return Math.max(0, Math.min(support.getBonusType().getCount(), intMap.size()));
```

Comment:
Previously it was using `getSupportLeft()`, which was using `details.totalSupport`, but now it's using `details.intMap.size()`. It's not clear why the latter is more correct. A comment in the code would be useful.